front/lib/api/assistant/global_agents.ts

```diff
@@ -88,15 +92,17 @@ async function _getHelperGlobalAgent(
   if (!owner) {
     throw new Error("Unexpected `auth` without `workspace`.");
   }
-  const model = !auth.isUpgraded()
-    ? {
-        providerId: GPT_3_5_TURBO_MODEL_CONFIG.providerId,
-        modelId: GPT_3_5_TURBO_MODEL_CONFIG.modelId,
-      }
-    : {
-        providerId: GPT_4_TURBO_MODEL_CONFIG.providerId,
-        modelId: GPT_4_TURBO_MODEL_CONFIG.modelId,
-      };
+  const modelConfiguration = auth.isUpgraded()
+    ? getLargeWhitelistedModel(owner)
+    : getSmallWhitelistedModel(owner);
+  if (!modelConfiguration) {
+    throw new Error("No whitelisted models were found for the workspace.");
```

Comment:
This can throw if there is no provider enabled which is not great. Let's return the assistant but with a disabled status instead to avoid 500s all over the place if that happens

Reply:
@spolu How can there be no provider enabled? We enforce at least one provider when admins manually want to modify the list. 
By default `whiteListedProviders` is `null`. But in this setup `whiteListedProviders` will be set to `MODEL_PROVIDER_IDS`.

Reply:
The enforcement is UI only and one day there will be ways to change that through poke. Generally speaking we should not assume anything about DB state and be able to handle gracefully if easy. Here it's easy we just need to set status to "disabled_by_admin"